src/ReplicatedStorage/Client/InterfaceComponents/Toolbar/ToolbarItem.lua

```diff
@@ -0,0 +1,234 @@
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local Packages = ReplicatedStorage:WaitForChild("Packages")
+
+local Fusion = require(Packages.Fusion)
+local WrappedFusion = require(Packages.FusionUtils).Wrapped
+
+local UIComponents = script.Parent.Parent
+local ClientSrc = UIComponents.Parent
+
+local SharedInterfaceState = require(ClientSrc.SharedInterfaceState)
+local Button = require(UIComponents.Button)
+
+local New = WrappedFusion.New
+local Statify = WrappedFusion.Statify
+local Children = Fusion.Children
+local Value = Fusion.Value
+local Computed = Fusion.Computed
+local Spring = Fusion.Spring
+local Out = Fusion.Out
+
+local function ToolbarItem(props)
+	-- Basic states
+	local hovering = props.Hovering or Value(false)
+	local pressing = props.Pressing or Value(false)
+	local visible = Statify(if props.Visible ~= nil then props.Visible else true)
+	local selected = Statify(props.Selected or false)
+
+	-- Tool data states
+	local inputName = Statify(props.InputName)
+	local toolName = Statify(props.ToolName)
+
+	-- Sizing states
+	local inputNameTextBounds = Value(Vector2.new())
+	local toolNameTextBounds = Value(Vector2.new())
+	local absoluteButtonSize = Value(Vector2.new())
+
+
+	-- Size States
+	local paddingSize = Computed(function()
+		local currentAbsoluteButtonSize = absoluteButtonSize:get()
+
+		if currentAbsoluteButtonSize == nil then
+			return 0
+		else
+			return currentAbsoluteButtonSize.Y / 10
+		end
```

Comment:
Could this perhaps be an if expression?

Reply:
I'm not sure it would even look cleaner - try it and see what you think - feel free to dismiss if it looks worse.